src/platform_impl/linux/x11/window.rs

```diff
@@ -121,9 +121,8 @@ impl UnownedWindow {
         pl_attribs: PlatformSpecificWindowBuilderAttributes,
     ) -> Result<UnownedWindow, RootOsError> {
         let xconn = &event_loop.xconn;
-        let root = if let Some(id) = pl_attribs.parent_id {
-            // WindowId is XID under the hood which doesn't exceed u32, so this conversion is lossless
-            u64::from(id) as _
+        let root = if let Some(RawWindowHandle::Xlib(handle)) = window_attrs.parent_window {
+            handle.window
```

Comment:
`WindowBuilderExtX11::with_parent` takes `WindowId`. I assume it only handles Xlib?
I only saw Xcb in raw_window_handle_04 usage.

Reply:
It would probably be fine to accept both. The `window` and `visual_id` are both the same, except Xlib has (likely for historical reasons) these values typed as `c_ulong`, even though I'm fairly certain both values are guaranteed to be 32-bit at the protocol level.

Reply:
Alright! I update with a Xcb arm to match.